src/AnalyzeCli.cpp

```diff
@@ -67,7 +68,21 @@ static void reportWarning(ReportFormat format, const char* name, const Luau::Lin
     report(format, name, warning.location, Luau::LintWarning::getName(warning.code), warning.text.c_str());
 }
 
-static bool analyzeFile(Luau::Frontend& frontend, const char* name, ReportFormat format, bool annotate)
+static bool isIgnoredFile(const std::filesystem::path& rootUriPath, const std::filesystem::path& path, std::vector<std::string>& ignoreGlobPatterns) {
+    auto relativePath = path.lexically_relative(rootUriPath).generic_string(); // HACK: we convert to generic string so we get '/' separators
+
+    //luau analyze returns relative path for files that are to be analyzed
+    if (relativePath.empty())
+        relativePath = path;
```

Comment:
When will this get hit? I can't think of when relativePath would be empty

Reply:
This gets hit when the original path is already relative to the root path. The path returned by the file resolver is relative to the root path for the files that are passed into the analyze command, however, for files that contain errors outside of the files that you have specified in the command, the path is absolute. This check was initially introduced to handle the case where the file you're trying to analyze is also a file you're trying to ignore (e.g `luau-lsp analyze src/shared/test.lua --ignoreGlobs=src/shared/test.lua`). In this case, the relative path conversion returned an empty string since the path was already relative to the root path. Which I'm not sure is a case we should be solving for in the first place, but I could imagine a situation where someone might want to see lints, but not errors. Thoughts?

Reply:
That inconsistency is somewhat weird (files passed are relative but the rest are not).
I think this is fine for now